File: solutions/205-isomorphic-strings/isomorphic-strings.cpp
class Solution {
public:
    bool isIsomorphic(string s, string t) {
        int n=s.length();
        if(n!=t.length()) return false;
        unordered_map<char,char> mp;
        unordered_set<char> set;
        for(int i=0;i<n;i++){
            if(!mp.count(s[i])){
                if(set.count(t[i])){
                    return false;
                }
                mp[s[i]]=t[i];
                set.insert(t[i]);
            }else if(mp[s[i]]!=t[i]){
                return false;
            }
        }
        return true;
    }
};

File: solutions/205-isomorphic-strings/isomorphic-strings.java
class Solution {
    public boolean isIsomorphic(String s, String t) {
        int n1=s.length();
        int n2=t.length();
        if(n1!=n2){
            return false;
        }
        Map<Character,Character> mp=new HashMap<Character,Character>();
        Set<Character> set=new HashSet<>();
        int i=0;
        while(i<n1){
            Character x=s.charAt(i);
            Character y=t.charAt(i);
            if((mp.containsKey(x) && !mp.get(x).equals(y)) || (!mp.containsKey(x) && set.contains(y))){
                return false;
            }
            mp.put(x,y);
            set.add(y);
            // mp.put(t.charAt(i),s.charAt(i));
            i++;
        }
        return true;
    }
}